src/mbgl/map/map.cpp

```diff
@@ -352,6 +352,8 @@ void Map::Impl::loadStyleJSON(const std::string& json) {
 
     // force style cascade, causing all pending transitions to complete.
     style->cascade(Clock::now(), mode);
+    
+    view.notifyMapChange(MapChangeDidFinishLoadingStyle);
```

Comment:
@tmpsantos to keep things clean, I've added a method to the observer on Style and kept the signal emitting in Map.